### PR TITLE
8139348: Deprecate 3DES and RC4 in Kerberos

### PR DESCRIPTION
**Pull request by wangweij**
Deprecate des3-hmac-sha1 (etype 16) and rc4-hmac (etype 23). User can add "allow_weak_crypto = true" in krb5.conf to re-enable them (plus the DES-based etypes deprecated long ago).

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8139348](https://bugs.openjdk.java.net/browse/JDK-8139348): Deprecate 3DES and RC4 in Kerberos

### Reviewers
 * [Sean Mullan](https://openjdk.java.net/census#mullan) (@seanjmullan - **Reviewer**) ⚠️ Review applies to 146bc47a2170fd2452d26fee863f7d344f9dd746

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2701/head:pull/2701`
`$ git checkout pull/2701`

### CODE REVIEW

File: test/jdk/sun/security/krb5/auto/NewSalt.java [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
 /*
  * @test
- * @bug 6960894 8194486
+ * @bug 6960894 8194486 8139348
```
Comment (seanjmullan) [inline test/jdk/sun/security/krb5/auto/NewSalt.java:26]: I don't know if there are any rules or best practices about this, but I usually don't put a bugid on a test if it isn't specifically testing what this bug is about.

**Comment by seanjmullan** [comment]
Is there a test that checks that the weak algorithms are actually disabled? I wasn't sure if I saw anything or maybe that is another test that you didn't have to modify?

File: test/jdk/sun/security/krb5/auto/NewSalt.java (hunk 1 above)
Comment (wangweij) [reply test/jdk/sun/security/krb5/auto/NewSalt.java:26]: OK, I'll remove it.

**Comment by wangweij** [comment]
> Is there a test that checks that the weak algorithms are actually disabled? I wasn't sure if I saw anything or maybe that is another test that you didn't have to modify?

Actually there's no. I can add all weak etypes into `onlythree.conf` and they should be ignored.

**Comment by wangweij** [comment]
Updated tests. There is a `weakcrypto.conf` file which has been useless for a long time since `WeakCrypto.java` generates krb5.conf on the fly.

**Comment by wangweij** [comment]
Please also review the release note at https://bugs.openjdk.java.net/browse/JDK-8262335.

File: test/jdk/sun/security/krb5/auto/W83.java [hunk 2]
```diff
@@ -23,16 +23,16 @@
 
 /*
  * @test
- * @bug 6932525 6951366 6959292 8194486
+ * @bug 6932525 6951366 6959292 8194486 8139348
```
Comment (seanjmullan) [inline test/jdk/sun/security/krb5/auto/W83.java:26]: I don't think you should add the bugid here since it isn't really about testing this change.

**Review by seanjmullan — APPROVED** [review]

File: test/jdk/sun/security/krb5/auto/W83.java (hunk 2 above)
Comment (wangweij) [reply test/jdk/sun/security/krb5/auto/W83.java]: OK, so as long as a test is not meant to test the change for a bug, then its bugid should not appear there.

**Pull request closed without merging**